scout/adapter/mongo/variant.py

```diff
@@ -689,39 +689,43 @@ def delete_variants(self, case_id, variant_type, category=None):
         result = self.variant_collection.delete_many(query)
         LOG.info("{0} variants deleted".format(result.deleted_count))
 
-    def overlapping(self, variant_obj, limit=30):
+    def hgnc_overlapping(self, variant_obj: dict, limit: int = None) -> Iterable[Dict]:
         """Return overlapping variants.
 
-        Look at the genes that a variant overlaps to.
-        Then return all variants that overlap these genes.
+        Look at the genes that a variant overlaps, then return all variants that overlap these genes.
 
-        If variant_obj is sv it will return the overlapping snvs and oposite
-        There is a problem when SVs are huge since there are to many overlapping variants.
+        The operation is slightly different depending on the category of the variants that we want to collect.
+        If variant_obj is an SV it will return the hgnc_id matching SVs, SNVs, and MEIs but
+        for SNVs we will only return the SVs and MEIs since the genmod compounds are way better.
 
-        Args:
-            variant_obj(dict)
+        Do not return the present variant as matching.
 
-        Returns:
-            variants(iterable(dict))
+        limit: A maximum count of returned variants is introduced: mainly this is a problem when SVs are huge since there can be many genes and overlapping variants.
+               We sort to offer the LIMIT most severe overlapping variants.
         """
-        # This is the category of the variants that we want to collect
-        category = "snv" if variant_obj["category"] == "sv" else "sv"
+        category = (
+            {"$in": ["sv", "mei"]}
+            if variant_obj["category"] == "snv"
+            else {"$in": ["sv", "snv", "mei", "cancer", "cancer_sv"]}
```

Comment:
We could do multiple as well, with cancer-cancer_sv on their own, but we usually don't have overlaps between these in the same case. If we do, I only think this function will help.